ibis/expr/operations/core.py

```diff
@@ -37,17 +37,36 @@ def equals(self, other):
     def to_expr(self):
         ...
 
-    # TODO(kszucs): introduce a HasName schema, or NamedValue with a .name
-    # abstractproperty
-    def resolve_name(self):
-        raise ExpressionError(f'Expression is not named: {type(self)}')
 
-    def has_resolved_name(self):
```

Comment:
We can maintain a backward compatible API for another major release if that would be preferable, though we break internals heavily with the removal of intermediate expressions.

Reply:
This seems fine. I don't think there's much use of `has_resolved_name()` in the wild.